sphinx/builders/html/transforms.py

```diff
@@ -40,7 +40,10 @@ class KeyboardTransform(SphinxPostTransform):
 
     def run(self, **kwargs: Any) -> None:
         matcher = NodeMatcher(nodes.literal, classes=["kbd"])
-        for node in self.document.findall(matcher):  # type: nodes.literal
+        # this list must be pre-created as during iteration new nodes
+        # are added which match the condition in the NodeMatcher.
+        nodes_to_expand: List[nodes.literal] = list(self.document.findall(matcher))
+        for node in nodes_to_expand:  # type: nodes.literal
```

Comment:
Wrapping `findall()` with `list()` is a well-used technique in Sphinx. So no temporary variable and comments are needed, I think. How about this?
```suggestion
        for node in list(self.document.findall(matcher)):  # type: nodes.literal
```

Note: If you'd like to declare a temporary variable, please remove `# type` comment from this line. It's duplicated.

Reply:
Removed the temporary variable.